pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -361,6 +361,23 @@ public synchronized PinotResourceManagerResponse addInstance(Instance instance)
     }
   }
 
+  /**
+   * Update a given instance for the specified Instance ID
+   */
+  public synchronized PinotResourceManagerResponse updateInstance(String instanceIdToUpdate, Instance newInstance) {
+    List<String> instances = getAllInstances();
+    if (!instances.contains(instanceIdToUpdate)) {
+      return PinotResourceManagerResponse.failure("Instance " + instanceIdToUpdate + " does not exists");
+    } else {
+      InstanceConfig newConfig = InstanceUtils.toHelixInstanceConfig(newInstance);
+      if(!_helixZkManager.getHelixDataAccessor().setProperty(
+          _helixZkManager.getHelixDataAccessor().keyBuilder().instanceConfig(instanceIdToUpdate), newConfig)) {
```

Comment:
For performance and simplicity:
```suggestion
    if (getHelixInstanceConfig(instanceIdToUpdate) == null) {
      return PinotResourceManagerResponse.failure("Instance " + instanceIdToUpdate + " does not exists");
    } else {
      InstanceConfig newConfig = InstanceUtils.toHelixInstanceConfig(newInstance);
      if(!_helixDataAccessor.setProperty(_keyBuilder.instanceConfig(instanceIdToUpdate), newConfig)) {
```